src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -588,7 +588,38 @@ public OptionalLong indexOf(Predicate<? super T> predicate) {
             return a == null ? null : StreamEx.of(Arrays.spliterator(a));
         });
     }
-    
+
+    /**
+     * Returns a stream consisting of the results of replacing each element of
+     * this stream with the contents of the optional value produced by applying
+     * the provided mapping function to each element.
+     *
+     * <p>
+     * If the mapping function returns {@link Optional#empty()}, the original
+     * value will be removed from the resulting stream.
```

Comment:
Documentation should state that if mapping function returns `null`, an NPE might be thrown during terminal operation execution.